archiver/integrity.py

```diff
@@ -4,19 +4,29 @@
 
 from . import helpers
 from .constants import COMPRESSED_ARCHIVE_SUFFIX, ENCRYPTED_ARCHIVE_SUFFIX, \
-    LISTING_SUFFIX
+    LISTING_SUFFIX, HASH_SUFFIX, TAR_HASH_SUFFIX, COMPRESSED_ARCHIVE_HASH_SUFFIX, \
+    ENCRYPTED_ARCHIVE_HASH_SUFFIX
 from .extract import extract_archive
 from .listing import parse_tar_listing
 
 
 def check_integrity(source_path, deep_flag=False, threads=None, work_dir=None):
+
     archives_with_hashes = get_archives_with_hashes_from_path(source_path)
     is_encrypted = helpers.path_target_is_encrypted(source_path)
 
     logging.info("Starting integrity check on: " + source_path.as_posix())
 
     check_result = shallow_integrity_check(archives_with_hashes, workers=threads)
 
+    if source_path.is_dir():
+        integrity_result = check_archive_list_integrity(source_path)
+        if not integrity_result:
+            logging.error(
+                "Integrity check unsuccessful. Files missing in archive.")
+        check_result = check_result and integrity_result
+
```

Comment:
Isn't there an `else` missing calling check_archive_part_integrity(source_path) in case source_path is a part file?

Reply:
This is a good question. I wasn't quite sure what the intended behaviour was for an integrity check on a single file. So far I have interpreted it as just checking the completeness and sanity of that single file, so the completeness check on the directory is not performed. One could change this behaviour and check the parent directory of a single give file. Would that be what you would expect?

Also, it is silently assumed that the `source_path` is indeed the source directory of a given archive. Any other sub-directory would fail the check.

Reply:
The integrity check can be called for a single part file, e.g. my-archive.part2.tar.lz (see also integrity.get_archives_with_hashes_from_path dispatching the cases), that is, I'm only interested in this part2. For consistency, I would expect, that in this case it is also verified, that the various files associated to this part2 exist. So in an else clause, the check_archive_part_integrity could be called with `source_path`.

`source_path` is not a very good name by the way (I think it may come from some other code I copy pasted), feel free to rename to `archive_path` or similar.

Reply:
OK, thanks for clarifying this. I think your suggestion makes sense and I have implemented it this way now. I will keep the renaming of `source_path` for a later PR to not unnecessarily complicate things now.